app/controllers/post_controller.rb

```diff
@@ -41,14 +42,18 @@ def submit
   end
 
   private
-    def validate_submission_parameters(title, author, markdown_text)
+    def validate_submission_parameters(title, author, markdown_text, hero)
       validation_message = nil
       if title.empty?
         validation_message = 'A post cannot be submited with a blank title.'
       elsif author.empty?
         validation_message = 'A post cannot be submited without an author.'
       elsif markdown_text.empty?
-        validation_message = 'A post cannot be submited with no markdown content.'
+        validation_message = 'A post cannot be submited with no markdown content.' 
+      elsif !hero.empty? && !(hero =~ URI::regexp)
+        validation_message = 'The background image must be a valid URL.'
+      elsif !PostService.is_valid_hero(hero)
```

Comment:
Same thing like above. Please add a test case for when the hero url is not image.